mmf/trainers/lightning_core/loop_callback.py

```diff
@@ -2,35 +2,196 @@
 
 import logging
 
+import torch
+from mmf.common.meter import Meter
 from mmf.common.registry import registry
+from mmf.common.report import Report
+from mmf.trainers.core.reporting import TrainerReportingMixin
+from mmf.utils.logger import calculate_time_left, summarize_report
+from mmf.utils.timer import Timer
 from pytorch_lightning.callbacks.base import Callback
 
 
 logger = logging.getLogger(__name__)
 
 
-class LightningLoopCallback(Callback):
+class LightningLoopCallback(Callback, TrainerReportingMixin):
     def __init__(self, lightning_trainer):
         super().__init__()
         self.lightning_trainer = lightning_trainer
+        self.trainer_config = lightning_trainer.trainer_config
+        self.training_config = lightning_trainer.training
+
+        # for logging
+        self.total_timer = Timer()
+        self.snapshot_iterations = len(lightning_trainer.data_module.val_loader)
+        self.snapshot_iterations //= self.training_config.batch_size
```

Comment:
could you set these attributes afterwards in on_train_start or on_test_start? there you have access to the trainer, so you don't need to pass it as an input to the constructor here. otherwise you need to pass the trainer into the callback constructor and the callback into the trainer's constructor. not sure how that's working